app/controllers/resources_controller.rb

```diff
@@ -3,84 +3,85 @@
 require 'base64'
 
 class ResourcesController < ApplicationController
+  class BlobError < StandardError; end
+
   before_action :authorize_request
 
   # POST /objects
-  # rubocop:disable Metrics/AbcSize
-  # rubocop:disable Metrics/MethodLength
   def create
     authorize! :resource
-    begin
-      response_cocina_obj = Dor::Services::Client.objects.register(params: cocina_model)
-    rescue Dor::Services::Client::UnauthorizedResponse => e
-      return render build_error('Error registering object with dor-services-app', e, '500', true)
-    rescue Dor::Services::Client::UnexpectedResponse => e
-      return render build_error('Error registering object with dor-services-app', e, '502')
-    rescue Dor::Services::Client::ConnectionFailed => e
-      return render build_error('Unable to reach dor-services-app', e, '504', true)
-    end
 
-    # Doing this here rather than IngestJob so that have accurate timestamp for milestone.
     begin
-      workflow_client.create_workflow_by_name(response_cocina_obj.externalIdentifier, 'registrationWF', version: 1)
-    rescue Dor::WorkflowException => e
-      return render build_error('Error creating registrationWF with workflow-service', e, '502')
+      request_dro = cocina_model(params.except(:action, :controller, :resource, :accession).to_unsafe_h)
+    rescue BlobError => e
+      # Returning 500 because not clear whose fault it is.
+      return render build_error('500', e, 'Error matching uploading files to file parameters.')
     end
-
-    result = BackgroundJobResult.create
-    IngestJob.perform_later(druid: response_cocina_obj.externalIdentifier,
-                            filesets: params[:structural].to_unsafe_h.fetch(:contains, []),
+    result = BackgroundJobResult.create(output: {})
+    IngestJob.perform_later(model_params: JSON.parse(request_dro.to_json), # Needs to be sidekiq friendly serialization
```

Comment:
I think this is fine, but you could have also done `model_params: request_dro.to_json`, right?

Reply:
Yes, which would have passed a string instead of a hash.